==11627== Memcheck, a memory error detector
==11627== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==11627== Using Valgrind-3.10.0.SVN and LibVEX; rerun with -h for copyright info
==11627== Command: ./TestInteger
==11627== 
Running main() from gtest_main.cc
[==========] Running 76 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 76 tests from Integer
[ RUN      ] Integer.shift_left_digits
[       OK ] Integer.shift_left_digits (35 ms)
[ RUN      ] Integer.shift_left_digits2
[       OK ] Integer.shift_left_digits2 (10 ms)
[ RUN      ] Integer.shift_left_digits3
[       OK ] Integer.shift_left_digits3 (5 ms)
[ RUN      ] Integer.shift_left_digits4
[       OK ] Integer.shift_left_digits4 (5 ms)
[ RUN      ] Integer.shift_right_digits
[       OK ] Integer.shift_right_digits (5 ms)
[ RUN      ] Integer.shift_right_digits2
[       OK ] Integer.shift_right_digits2 (5 ms)
[ RUN      ] Integer.shift_right_digits3
[       OK ] Integer.shift_right_digits3 (5 ms)
[ RUN      ] Integer.shift_right_digits4
[       OK ] Integer.shift_right_digits4 (5 ms)
[ RUN      ] Integer.plus_digits
==11627== Invalid write of size 4
==11627==    at 0x4300B7: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409157: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c5eac8 is 0 bytes after a block of size 24 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409060: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
==11627== Invalid read of size 2
==11627==    at 0x4C2F647: memcpy@GLIBC_2.2.5 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43C10C: int* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<int>(int const*, int const*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43A734: int* std::__copy_move_a<false, int*, int*>(int*, int*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x438AE2: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::__copy_move_a2<false, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43607C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::copy<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F5D4: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x4304E4: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409157: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c5eac8 is 0 bytes after a block of size 24 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409060: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
==11627== Invalid write of size 4
==11627==    at 0x438B63: __gnu_cxx::__enable_if<std::__is_scalar<int>::__value, void>::__type std::__fill_a<int*, int>(int*, int*, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x436170: void std::fill<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int>(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F635: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x4304E4: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409157: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c5eac8 is 0 bytes after a block of size 24 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409060: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
[       OK ] Integer.plus_digits (12 ms)
[ RUN      ] Integer.plus_digits2
==11627== Invalid write of size 4
==11627==    at 0x4300B7: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409A75: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c5f3cc is 0 bytes after a block of size 12 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40997E: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
==11627== Invalid read of size 2
==11627==    at 0x4C2F638: memcpy@GLIBC_2.2.5 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43C10C: int* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<int>(int const*, int const*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43A734: int* std::__copy_move_a<false, int*, int*>(int*, int*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x438AE2: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::__copy_move_a2<false, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43607C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::copy<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F5D4: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x4304E4: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x409A75: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c5f3ce is 2 bytes after a block of size 12 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40997E: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
[       OK ] Integer.plus_digits2 (6 ms)
[ RUN      ] Integer.plus_digits3
==11627== Invalid write of size 4
==11627==    at 0x4300B7: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40A393: Integer_plus_digits3_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c5fbac is 0 bytes after a block of size 12 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40A29C: Integer_plus_digits3_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
[       OK ] Integer.plus_digits3 (8 ms)
[ RUN      ] Integer.plus_digits4
==11627== Invalid write of size 4
==11627==    at 0x4300B7: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40ACB1: Integer_plus_digits4_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c6038c is 0 bytes after a block of size 12 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40ABBA: Integer_plus_digits4_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
[       OK ] Integer.plus_digits4 (7 ms)
[ RUN      ] Integer.plus_digits5
==11627== Invalid write of size 4
==11627==    at 0x4300B7: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40B649: Integer_plus_digits5_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c60b7c is 0 bytes after a block of size 12 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40B4D8: Integer_plus_digits5_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
[       OK ] Integer.plus_digits5 (6 ms)
[ RUN      ] Integer.plus_digits6
==11627== Invalid write of size 4
==11627==    at 0x4300B7: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40C04A: Integer_plus_digits6_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c6137c is 0 bytes after a block of size 12 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40BE9C: Integer_plus_digits6_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
[       OK ] Integer.plus_digits6 (7 ms)
[ RUN      ] Integer.plus_digits7
==11627== Invalid write of size 4
==11627==    at 0x4300B7: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40C9E4: Integer_plus_digits7_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c61df0 is 0 bytes after a block of size 336 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x40C8B0: Integer_plus_digits7_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
[       OK ] Integer.plus_digits7 (7 ms)
[ RUN      ] Integer.plus_digits8
[       OK ] Integer.plus_digits8 (6 ms)
[ RUN      ] Integer.minus_digits
[       OK ] Integer.minus_digits (26 ms)
[ RUN      ] Integer.minus_digits2
[       OK ] Integer.minus_digits2 (6 ms)
[ RUN      ] Integer.minus_digits3
[       OK ] Integer.minus_digits3 (5 ms)
[ RUN      ] Integer.minus_digits4
[       OK ] Integer.minus_digits4 (6 ms)
[ RUN      ] Integer.minus_digits5
[       OK ] Integer.minus_digits5 (6 ms)
[ RUN      ] Integer.minus_digits6
[       OK ] Integer.minus_digits6 (6 ms)
[ RUN      ] Integer.multiplies_digits
this is temp 18
this is what is in product sum 1 8
top shift amount should be zero and is 0
size of topshiftproduct should be 2 and is 2
==11627== Invalid write of size 4
==11627==    at 0x436FE4: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x4315AA: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411487: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x404FCF: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c69f58 is 0 bytes after a block of size 8 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x431284: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411487: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
==11627== Invalid write of size 4
==11627==    at 0x438B63: __gnu_cxx::__enable_if<std::__is_scalar<int>::__value, void>::__type std::__fill_a<int*, int>(int*, int*, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x436170: void std::fill<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int>(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F635: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x437411: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x4315AA: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411487: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45CCAD: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c69f58 is 0 bytes after a block of size 8 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x431284: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411487: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
this is the running total in the inner loop: 
1
8
this is temp 6
this is what is in product sum 0 6
top shift amount should be zero and is 1
size of topshiftproduct should be 2 and is 3
this is the running total in the inner loop: 
0
7
8
this is temp 4
this is what is in product sum 0 4
top shift amount should be zero and is 2
size of topshiftproduct should be 2 and is 4
this is the running total in the inner loop: 
0
4
7
8
this is the running total in the outer loop: 
0
4
7
8
this is temp 9
this is what is in product sum 0 9
top shift amount should be zero and is 0
size of topshiftproduct should be 2 and is 2
this is the running total in the inner loop: 
0
4
7
8
0
this is temp 3
this is what is in product sum 0 3
top shift amount should be zero and is 1
size of topshiftproduct should be 2 and is 3
this is the running total in the inner loop: 
0
5
0
8
0
0
this is temp 2
this is what is in product sum 0 2
top shift amount should be zero and is 2
size of topshiftproduct should be 2 and is 4
this is the running total in the inner loop: 
0
5
2
8
0
0
0
this is the running total in the outer loop: 
0
5
2
8
0
0
0
0
5
2
8
0
0
0
thats what was in total at the end
==11627== Invalid write of size 8
==11627==    at 0x4C2F5F3: memcpy@GLIBC_2.2.5 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43C10C: int* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<int>(int const*, int const*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43A734: int* std::__copy_move_a<false, int*, int*>(int*, int*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x438AE2: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::__copy_move_a2<false, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43607C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::copy<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x432AB0: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411487: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c69dd0 is 0 bytes after a block of size 16 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411390: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
==11627== Invalid write of size 2
==11627==    at 0x4C2F63B: memcpy@GLIBC_2.2.5 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43C10C: int* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<int>(int const*, int const*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43A734: int* std::__copy_move_a<false, int*, int*>(int*, int*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x438AE2: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::__copy_move_a2<false, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43607C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::copy<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x432AB0: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411487: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==  Address 0x5c69dd8 is 8 bytes after a block of size 16 alloc'd
==11627==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11627==    by 0x43A5FD: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43887C: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x43897C: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x435EFE: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x42F47F: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x411390: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x469BA2: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C5BC: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C653: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C754: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627==    by 0x45C9CC: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==11627== 
TestInteger.c++:224: Failure
Value of: equal(z.begin(), z.end(), a.begin())
  Actual: false
Expected: true
[  FAILED  ] Integer.multiplies_digits (66 ms)
[ RUN      ] Integer.divides_digits
[       OK ] Integer.divides_digits (9 ms)
[ RUN      ] Integer.divides_digits2
[       OK ] Integer.divides_digits2 (6 ms)
[ RUN      ] Integer.divides_digits3
[       OK ] Integer.divides_digits3 (6 ms)
[ RUN      ] Integer.divides_digits4
[       OK ] Integer.divides_digits4 (6 ms)
[ RUN      ] Integer.divides_digits5
[       OK ] Integer.divides_digits5 (6 ms)
[ RUN      ] Integer.divides_digits6
[       OK ] Integer.divides_digits6 (7 ms)
[ RUN      ] Integer.equal_to
[       OK ] Integer.equal_to (15 ms)
[ RUN      ] Integer.equal_to2
[       OK ] Integer.equal_to2 (6 ms)
[ RUN      ] Integer.equal_to3
[       OK ] Integer.equal_to3 (5 ms)
[ RUN      ] Integer.equal_to4
[       OK ] Integer.equal_to4 (6 ms)
[ RUN      ] Integer.equal_to5
[       OK ] Integer.equal_to5 (5 ms)
[ RUN      ] Integer.equal_to6
[       OK ] Integer.equal_to6 (9 ms)
[ RUN      ] Integer.equal_to7
[       OK ] Integer.equal_to7 (4 ms)
[ RUN      ] Integer.equal_to8
[       OK ] Integer.equal_to8 (5 ms)
[ RUN      ] Integer.equal_to9
[       OK ] Integer.equal_to9 (4 ms)
[ RUN      ] Integer.less_than
[       OK ] Integer.less_than (9 ms)
[ RUN      ] Integer.less_than2
[       OK ] Integer.less_than2 (5 ms)
[ RUN      ] Integer.less_than3
[       OK ] Integer.less_than3 (3 ms)
[ RUN      ] Integer.less_than4
[       OK ] Integer.less_than4 (4 ms)
[ RUN      ] Integer.less_than5
[       OK ] Integer.less_than5 (3 ms)
[ RUN      ] Integer.less_than6
[       OK ] Integer.less_than6 (4 ms)
[ RUN      ] Integer.less_than7
[       OK ] Integer.less_than7 (4 ms)
[ RUN      ] Integer.less_than8
[       OK ] Integer.less_than8 (3 ms)
[ RUN      ] Integer.less_than9
[       OK ] Integer.less_than9 (4 ms)
[ RUN      ] Integer.less_than10
[       OK ] Integer.less_than10 (3 ms)
[ RUN      ] Integer.less_than11
[       OK ] Integer.less_than11 (3 ms)
[ RUN      ] Integer.less_than12
[       OK ] Integer.less_than12 (2 ms)
[ RUN      ] Integer.less_than13
[       OK ] Integer.less_than13 (2 ms)
[ RUN      ] Integer.negation
[       OK ] Integer.negation (9 ms)
[ RUN      ] Integer.negation2
[       OK ] Integer.negation2 (4 ms)
[ RUN      ] Integer.negation3
[       OK ] Integer.negation3 (3 ms)
[ RUN      ] Integer.negation4
[       OK ] Integer.negation4 (3 ms)
[ RUN      ] Integer.negation5
[       OK ] Integer.negation5 (3 ms)
[ RUN      ] Integer.abs1
[       OK ] Integer.abs1 (4 ms)
[ RUN      ] Integer.abs2
[       OK ] Integer.abs2 (3 ms)
[ RUN      ] Integer.abs3
[       OK ] Integer.abs3 (3 ms)
[ RUN      ] Integer.abs4
[       OK ] Integer.abs4 (4 ms)
[ RUN      ] Integer.ostream1
[       OK ] Integer.ostream1 (7 ms)
[ RUN      ] Integer.ostream2
[       OK ] Integer.ostream2 (3 ms)
[ RUN      ] Integer.ostream3
[       OK ] Integer.ostream3 (3 ms)
[ RUN      ] Integer.ostream4
[       OK ] Integer.ostream4 (4 ms)
[ RUN      ] Integer.ostream5
[       OK ] Integer.ostream5 (4 ms)
[ RUN      ] Integer.ostream6
[       OK ] Integer.ostream6 (3 ms)
[ RUN      ] Integer.left_shift_equals
[       OK ] Integer.left_shift_equals (4 ms)
[ RUN      ] Integer.left_shift_equals2
[       OK ] Integer.left_shift_equals2 (4 ms)
[ RUN      ] Integer.left_shift_equals3
[       OK ] Integer.left_shift_equals3 (6 ms)
[ RUN      ] Integer.left_shift_equals4
[       OK ] Integer.left_shift_equals4 (3 ms)
[ RUN      ] Integer.left_shift_equals5
[       OK ] Integer.left_shift_equals5 (3 ms)
[ RUN      ] Integer.right_shift_equals
[       OK ] Integer.right_shift_equals (3 ms)
[ RUN      ] Integer.right_shift_equals2
[       OK ] Integer.right_shift_equals2 (3 ms)
[ RUN      ] Integer.right_shift_equals3
[       OK ] Integer.right_shift_equals3 (4 ms)
[ RUN      ] Integer.right_shift_equals4
[       OK ] Integer.right_shift_equals4 (2 ms)
[ RUN      ] Integer.right_shift_equals5
[       OK ] Integer.right_shift_equals5 (3 ms)
[----------] 76 tests from Integer (509 ms total)

[----------] Global test environment tear-down
[==========] 76 tests from 1 test case ran. (537 ms total)
[  PASSED  ] 75 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Integer.multiplies_digits

 1 FAILED TEST
==11627== 
==11627== HEAP SUMMARY:
==11627==     in use at exit: 0 bytes in 0 blocks
==11627==   total heap usage: 2,122 allocs, 2,122 frees, 201,395 bytes allocated
==11627== 
==11627== All heap blocks were freed -- no leaks are possible
==11627== 
==11627== For counts of detected and suppressed errors, rerun with: -v
==11627== ERROR SUMMARY: 43 errors from 14 contexts (suppressed: 0 from 0)
